test/backend/wayland/conftest.py

```diff
@@ -27,6 +31,22 @@ def wayland_environment(outputs):
     yield env
 
 
+@pytest.fixture(scope="function")
+def wmanager(request, wayland_session):
+    """
+    This replicates the `manager` fixture except that the Wayland backend is hard-coded.
+    We cannot parametrize the `backend_name` fixture module-wide because it gets
+    parametrized by `pytest_generate_tests` in test/conftest.py and only one of these
+    parametrize calls can be used.
```

Comment:
Not quite following why this is needed. Can you explain for me?

Reply:
Ha yes, this is an annoying quirk of pytest. You'll see the same thing in `test/backend/x11/conftest.py` with an `xmanager` fixture. There is a good reason for it, to do with parametrising pytest via CLI (`--backend=X`), but I'm struggling to identify what the reason was. Let me look into this.

Reply:
If you make them all use `manager` then the x11 tests get attempted using the wayland backend, and vice-versa, via the `backend` fixture used by `manager`. Removing `xmanager` and `wmanager` and using `manager` causes this:
```
test/backend/test_backend.py::test_get_core_bad PASSED                                                 [  1%]
test/backend/wayland/test_window.py::test_info[1-wayland-BareConfig] PASSED                            [  3%]
test/backend/wayland/test_window.py::test_info[1-x11-BareConfig] FAILED                                [  5%]
```
Note the failed test is a wayland test parameterised by the x11 fixtures.

Reply:
Ok. Can't we mark the tests to skip in those scenarios?

Reply:
The only way (i think) that we can do that is something like this:
```py
@bare_config
def test_info(backend_name, manager):
    if backend_name != "wayland":
        raise pytest.skip()
    ...
```
We might be able to use some kind of decorator to automate this, perhaps, but it would need adding to all tests (perhaps not that difficult given they'll fail otherwise). However an undesirable side effect is that the output of pytest logs SKIPPED messages:
```
test/backend/test_backend.py::test_get_core_bad PASSED                                                                 [  3%]
test/backend/wayland/test_window.py::test_info[1-wayland-BareConfig] PASSED                                            [  6%]
test/backend/wayland/test_window.py::test_info[1-x11-BareConfig] SKIPPED                                               [  9%]
```
The current solution avoids this, but it does seem hacky and a bit strange (and is ultimately duplicate code that we ought to avoid). If you had any other ideas for how to handle this please do let me know!

Reply:
Stick with the current approach for now!